File: src/main/java/com/serdeliverance/cryptowallet/domain/Transaction.java
package com.serdeliverance.cryptowallet.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {

    private Long id;
    private Integer userId;
    private Integer cryptocurrencyId;
    private BigDecimal amount;
    private OperationType operationType;
    private String transactionDate;
}

File: src/main/kotlin/com/serdeliverance/cryptowallet/domain/Transaction.kt
package com.serdeliverance.cryptowallet.domain

import java.math.BigDecimal

data class Transaction(
    val id: Long,
    val userId: Int,
    val cryptoCurrencyId: Int,
    val amount: BigDecimal,
    val operationType: OperationType,
    val transactionDate: String
)